test/new-e2e/tests/apm/docker_test.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package apm
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awsdocker "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/docker"
+	"github.com/DataDog/test-infra-definitions/components/datadog/dockeragentparams"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+)
+
+type DockerFakeintakeSuite struct {
+	e2e.BaseSuite[environments.DockerHost]
+}
+
+func TestDockerFakeintakeSuite(t *testing.T) {
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awsdocker.Provisioner(awsdocker.WithAgentOptions(
+			dockeragentparams.WithAgentServiceEnvVariable("DD_APM_RECEIVER_SOCKET", pulumi.String("/var/run/datadog/apm.socket")), // Enable the UDS receiver in the trace-agent
+			dockeragentparams.WithAgentServiceEnvVariable("STATSD_URL", pulumi.String("unix:///var/run/datadog/dsd.socket")),      // Optional: UDS is more reliable for statsd metrics
+		))),
+	}
+	if devMode, err := strconv.ParseBool(devModeEnv); err == nil && devMode {
+		options = append(options, e2e.WithDevMode())
+	}
+	e2e.Run(t, &DockerFakeintakeSuite{}, options...)
+}
+
+func (s *DockerFakeintakeSuite) TestAPMDocker() {
+	s.Run("TraceAgentMetrics", s.TraceAgentMetrics)
+	s.Run("TracesOnUDS", s.TracesOnUDS)
+	s.Run("StatsOnUDS", s.StatsOnUDS)
+}
+
+func (s *DockerFakeintakeSuite) TraceAgentMetrics() {
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		expected := map[string]struct{}{
+			// "datadog.trace_agent.started":                         {}, // FIXME: this metric is flaky
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Is it possible that the trace-agent starts before the fakeintake is ready ? We should wait for the fakeintake to be ready, but if this is flaky, it sounds like a race condition between trace-agent start and fakeintake readiness

Reply:
That's a possibility, I'm also suspecting a more subtle bug like a race between the core-agent and the trace-agent, since these are dogstatsd metrics sent by the trace-agent through the core-agent. I'll file a bug to investigate this further.

Reply:
To exclude the first, you can dual-ship to `dddev` and see if you get it there (for the investigation)